File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/guild/GuildInformationsGeneralMessage.scala





















// Generated on 05/03/2013 10:34:50
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GuildInformationsGeneralMessage extends MessageDeserializer {
  val messageId = 5557

  protected trait Message extends BaseMessage with Serializable {
    val messageId = GuildInformationsGeneralMessage.messageId

    val enabled: Boolean
    val abandonnedPaddock: Boolean
    val level: UByte
    val expLevelFloor: Double
    val experience: Double
    val expNextLevelFloor: Double
    val creationDate: Int

    def serialize(buf: Buffer) {
      var flag1 = 0
      flag1 = BooleanByteWrapper.setFlag(flag1, 0, enabled)
      flag1 = BooleanByteWrapper.setFlag(flag1, 1, abandonnedPaddock)
      buf.writeUByte(flag1.toShort)
      buf.writeUByte(level)
      buf.writeDouble(expLevelFloor)
      buf.writeDouble(experience)
      buf.writeDouble(expNextLevelFloor)
      buf.writeInt(creationDate)
    }

  }

  protected def create(enabled: Boolean, abandonnedPaddock: Boolean, level: UByte, expLevelFloor: Double, experience: Double, expNextLevelFloor: Double, creationDate: Int): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val flag1 = buf.readUByte()
    val enabled = BooleanByteWrapper.getFlag(flag1, 0)
    val abandonnedPaddock = BooleanByteWrapper.getFlag(flag1, 1)
    val level = buf.readUByte()
    val expLevelFloor = buf.readDouble()
    val experience = buf.readDouble()
    val expNextLevelFloor = buf.readDouble()
    val creationDate = buf.readInt()

    create(enabled, abandonnedPaddock, level, expLevelFloor, experience, expNextLevelFloor, creationDate)
  }

}

object GuildInformationsGeneralMessage extends GuildInformationsGeneralMessage {
  case class MessageImpl(enabled: Boolean, abandonnedPaddock: Boolean, level: UByte, expLevelFloor: Double, experience: Double, expNextLevelFloor: Double, creationDate: Int) extends Message
  type Target = MessageImpl

  override def create(enabled: Boolean, abandonnedPaddock: Boolean, level: UByte, expLevelFloor: Double, experience: Double, expNextLevelFloor: Double, creationDate: Int) = MessageImpl(enabled, abandonnedPaddock, level, expLevelFloor, experience, expNextLevelFloor, creationDate)

  def apply(enabled: Boolean, abandonnedPaddock: Boolean, level: UByte, expLevelFloor: Double, experience: Double, expNextLevelFloor: Double, creationDate: Int) = create(enabled, abandonnedPaddock, level, expLevelFloor, experience, expNextLevelFloor, creationDate)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GuildInformationsGeneralMessage.java


// Generated on 05/08/2013 19:37:51
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GuildInformationsGeneralMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 5557;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public boolean enabled;
    public boolean abandonnedPaddock;
    public short level;
    public double expLevelFloor;
    public double experience;
    public double expNextLevelFloor;
    public int creationDate;
    
    public GuildInformationsGeneralMessage() { }
    
    public GuildInformationsGeneralMessage(boolean enabled, boolean abandonnedPaddock, short level, double expLevelFloor, double experience, double expNextLevelFloor, int creationDate) {
        this.enabled = enabled;
        this.abandonnedPaddock = abandonnedPaddock;
        this.level = level;
        this.expLevelFloor = expLevelFloor;
        this.experience = experience;
        this.expNextLevelFloor = expNextLevelFloor;
        this.creationDate = creationDate;
    }
    
    @Override
    public void serialize(Buffer buf) {
        short flag1 = 0;
        flag1 = BooleanByteWrapper.setFlag(flag1, 0, enabled);
        flag1 = BooleanByteWrapper.setFlag(flag1, 1, abandonnedPaddock);
        buf.writeUByte(flag1);
        buf.writeUByte(level);
        buf.writeDouble(expLevelFloor);
        buf.writeDouble(experience);
        buf.writeDouble(expNextLevelFloor);
        buf.writeInt(creationDate);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        short flag1 = buf.readUByte();
        enabled = BooleanByteWrapper.getFlag(flag1, 0);
        abandonnedPaddock = BooleanByteWrapper.getFlag(flag1, 1);
        level = buf.readUByte();
        if (level < 0 || level > 255)
            throw new RuntimeException("Forbidden value on level = " + level + ", it doesn't respect the following condition : level < 0 || level > 255");
        expLevelFloor = buf.readDouble();
        if (expLevelFloor < 0)
            throw new RuntimeException("Forbidden value on expLevelFloor = " + expLevelFloor + ", it doesn't respect the following condition : expLevelFloor < 0");
        experience = buf.readDouble();
        if (experience < 0)
            throw new RuntimeException("Forbidden value on experience = " + experience + ", it doesn't respect the following condition : experience < 0");
        expNextLevelFloor = buf.readDouble();
        if (expNextLevelFloor < 0)
            throw new RuntimeException("Forbidden value on expNextLevelFloor = " + expNextLevelFloor + ", it doesn't respect the following condition : expNextLevelFloor < 0");
        creationDate = buf.readInt();
        if (creationDate < 0)
            throw new RuntimeException("Forbidden value on creationDate = " + creationDate + ", it doesn't respect the following condition : creationDate < 0");
    }
    
}
